core/Network/TLS/Core.hs

```diff
@@ -212,3 +217,20 @@ terminate' ctx send err level desc reason = do
 -- | same as recvData but returns a lazy bytestring.
 recvData' :: MonadIO m => Context -> m L.ByteString
 recvData' ctx = recvData ctx >>= return . L.fromChunks . (:[])
+
+keyUpdate :: Context
+          -> (Context -> IO (Hash,Cipher,C8.ByteString))
+          -> (Context -> Hash -> Cipher -> C8.ByteString -> IO ())
+          -> IO ()
+keyUpdate ctx getState setState = do
+    (usedHash, usedCipher, applicationTrafficSecretN) <- getState ctx
+    let applicationTrafficSecretN1 = hkdfExpandLabel usedHash applicationTrafficSecretN "traffic upd" "" $ hashDigestSize usedHash
+    setState ctx usedHash usedCipher applicationTrafficSecretN1
+
+-- | Updating appication traffic secrets for TLS 1.3.
+updateKey :: Context -> IO ()
+updateKey ctx = do
+    tls13 <- tls13orLater ctx
+    when tls13 $ do
+        sendPacket13 ctx $ Handshake13 [KeyUpdate13 UpdateRequested]
+        keyUpdate ctx getTxState setTxState
```

Comment:
In TLS 1.3 specification, both Tx and Rx are updated. There is no way to do update one direction only.

Reply:
Done. `updateKey :: Context -> IO Bool`

Reply:
About Tx/Rx, this is not my understanding. What happens if Alice sends "update_not_requested" to Bob? Update of traffic key Alice -> Bob only.

One use case for key update is to avoid reaching traffic limits of nonce-based cipher modes. The amount of traffic in both directions is not necessarily balanced.

Reply:
Good point. I added the check.